code/game/objects/mail.dm

```diff
@@ -0,0 +1,233 @@
+/obj/item/envelope
+	name ="broken letter"
+	desc = "We just got a letter, we just got a letter, we just got a letter- wonder who its from."
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mail_misc"
+	item_state = "paper"
+	drop_sound = 'sound/items/handling/paper_drop.ogg'
+	pickup_sound =  'sound/items/handling/paper_pickup.ogg'
+	var/list/possible_contents = list()
+	var/list/job_list = list()
+	var/mob/living/recipient
+	var/has_been_scanned = FALSE
+
+/obj/item/envelope/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is licking a sharp corner of the envelope. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(loc, 'sound/effects/-adminhelp.ogg', 50, TRUE, -1)
+	return BRUTELOSS
+
+/obj/item/envelope/attack_self(mob/user)
+	if(user.mind.current != recipient.mind.current)
```

Comment:
You should probably be checking minds are equal here, rather than the mob they inhabit. The way it is currently would prevent people who were cloned from opening their own mail.

You'll also want to account for this wherever you set the mind for mail as well.

Reply:
`mind.current` should be enough to cover it, no?

Reply:
`mind.current` will check the mobs that the minds are currently inhabiting. If this is assigned when the mail is created, and say cargo gets a backlog, it might fall out of sync with who's actually inhabiting the item.
If you get cloned, you aren't in the same mob, you're in a different one. Likewise, if you russian revolver yourself and get transplanted into a monkey, even though you have the same mind (and possibly even the same brain), you might not be able to open your mail.
This would also mean that you could open your former body's mail as an ERT, but to me it feels like it'd be a much nicer solution that would avoid some of the pitfalls this might cause.